index.html

```diff
@@ -0,0 +1,475 @@
+
+> The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+> NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and
+> "OPTIONAL" in this document are to be interpreted as described in
+> RFC 2119.
+
+### What this document isn't
+
+This document specifies only the structure and semantics of the CSDL. It's expected that the CSDL will be the output of a compilation process which composes subgraph schemas. The mechanicss of that process are not specified normatively here; a suggestion is provided in [Appendix: Basic Composition Algorithm](#appendix-suggested-composition-algorithm). Conforming implementations may choose any approach they like, so long as the result is conforming CSDL.
+
+## Example: Let's go to the moon
+
+*This section is non-normative.*
+
+We'll refer to this example throughout the document. The example consists of two example schemas—a `rockets` schema that holds data about rockets, and an `astronauts` schema that holds information about the people flying on them—along with a CSDL that composes them.
+
+```graphql
+#::[Rockets example service]
+#::a.not-csdl
+type Query {
+  rockets: [Rocket]!
+}
+
+type Rocket @key(fields: "id") {
+  id: String! 
+  name: String!
+  captain: Astronaut @provides(fields: "tripId")
+}
+
+extend type Astronaut @key(fields: "id") {
+  id: String! @external
+  tripId: String! @external
+  rocket: Rocket!
+}
+```
+
+```graphql
+#::[Astronauts example service]
+#::a.not-csdl
+type Query {
+  astronauts: [Astronaut]!
+}
+
+type Astronaut @key(fields: "id") {
+  id: String! 
+  name: String!
+  tripId: String!
+}
+
+extend type Rocket @key(fields: "id") {
+  id: String! @external
+  astronaut: Astronaut!
+}
+```
+
+This example is non-normative, and represents just one possible way to compose [rockets](#rockets) and [astronauts](#astronauts) into CSDL:
+
+```graphql
+#::[CSDL composing `rockets` and `astronauts`]
+schema @using(spec: "https://specs.apollo.dev/csdl/v1") {
+  query: Query
+}
+
+enum csdl_Graph {
+  ROCKETS    @csdl_endpoint(url: "https://rockets.example.com"),
+  ASTRONAUTS @csdl_endpoint(url: "https://astronauts.example.com"),
+}
+
+type Query {
+  rockets: [Rocket]!       @csdl_resolve(graph: ROCKETS)
+  astronauts: [Astronaut]! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Astronaut
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!     @csdl_resolve(graph: ASTRONAUTS)
+  name: String!   @csdl_resolve(graph: ASTRONAUTS)
+  tripId: String! @csdl_resolve(graph: ASTRONAUTS)
+  rocket: Rocket! @csdl_resolve(graph: ASTRONAUTS)
+}
+
+type Rocket 
+  @csdl_key(graph: ASTRONAUTS, repr: "{ id }")
+  @csdl_key(graph: ROCKETS, repr: "{ id }")
+{
+  id: String!           @csdl_resolve(graph: ROCKETS)
+  name: String!         @csdl_resolve(graph: ROCKETS)
+  astronaut: Astronaut! @csdl_resolve(graph: ASTRONAUTS)
+  captain: Astronaut    @csdl_resolve(graph: ASTRONAUTS,
```

Comment:
Shouldn't this be ROCKETS? (Maybe I'm confused — I'm thinking the correct graph is the one that declares the field in a non-`@external` way?)

Reply:
yeah, that was an error, changed the example